src/main/java/org/snomed/snowstorm/core/data/services/ConceptService.java

```diff
@@ -143,6 +151,63 @@ public Concept find(String id, List<LanguageDialect> languageDialects, BranchTim
 		return doFind(conceptIds, languageDialects, new BranchTimepoint(path), pageRequest);
 	}
 
+	public ConceptHistory loadConceptHistory(String conceptId, List<CodeSystemVersion> codeSystemVersions) {
+		Map<String, BranchCriteria> branchCriteria = new HashMap<>();
+		for (CodeSystemVersion codeSystemVersion : codeSystemVersions) {
+			String branchPath = codeSystemVersion.getBranchPath();
+			branchCriteria.put(branchPath, versionControlHelper.getBranchCriteria(branchPath));
+		}
+
+		Function<String, BoolQueryBuilder> defaultBoolQueryFunction = cId -> {
+			BoolQueryBuilder someReleaseBranch = boolQuery();
+			BoolQueryBuilder boolQueryBuilder = boolQuery();
+			boolQueryBuilder.must(someReleaseBranch);
+			boolQueryBuilder.must(existsQuery(Concept.Fields.EFFECTIVE_TIME));
+			boolQueryBuilder.should(termQuery(Concept.Fields.CONCEPT_ID, cId)); //Find for Concept, Description & Axiom (RefSetMember) documents
+			boolQueryBuilder.should(termQuery(Relationship.Fields.SOURCE_ID, cId)); //Find for Relationship documents
+
+			return boolQueryBuilder;
+		};
+		PageRequest pageRequest = PageRequest.of(0, codeSystemVersions.size() + 1);
+		ConceptHistory conceptHistory = new ConceptHistory(conceptId);
+		for (Map.Entry<ComponentType, Class<? extends DomainEntity<?>>> entrySet : DOCUMENTS.entrySet()) {
+			ComponentType componentType = entrySet.getKey();
+			Class<? extends DomainEntity<?>> document = entrySet.getValue();
```

Comment:
this is not a document, variable name could be `documentType` or something ?

Reply:
I have changed the variable name to documentType.